ibis/expr/types/arrays.py

```diff
@@ -426,23 +443,28 @@ def map(self, func: Callable[[ir.Value], ir.Value]) -> ir.ArrayValue:
         │ []                     │
         └────────────────────────┘
         """
-        name = next(iter(inspect.signature(func).parameters.keys()))
+        if isinstance(func, Deferred):
+            name = "_"
```

Comment:
```suggestion
            name = str(func)
```

That will render the deferred object in an expression-y way:

```
In [3]: from ibis.interactive import *

In [4]: _.a + 1
Out[4]: (_.a + 1)

In [5]: str(_.a + 1)
Out[5]: '(_.a + 1)'
```